### ISSUE TITLE
[Deployment Issue] Unable to Access "captain.captain.localhost" or any other app "appname.captain.localhost" 

### ISSUE THREAD

**Issue by juanferr** [caprover/caprover#1528]
Hello,

Im installed caprover in a proxmox LXC docker instance. i followed the guidelines presented here in the [docs](https://caprover.com/docs/run-locally.html) 

everything seemed fine as im able to access caprover via` 192.x.x.x:3000`. i set the domain to `caprover.localhost `a per the instructions however, im unable to access the app via `appname.captain.localhost.` im also unable to access `captain.captain.localhost`

a nslookup pn` captain.captain.localhost` outputs the correct ip and they match my correct IP
```
Server:         192.X.X.X
Address:        192.X.X.X#53

Name:   captain.captain.localhost
Address: 192.X.X.X
```

the logs dont really say much  docker service logs captain-captain --since 15m

i tried `curl -v appname.captain.localhost` for example `bitwarden.captain.localhost` and i can see a `502` error just not sure where/what could be causing that issue.

thanks!

**Comment by githubsaturn** [comment]
What is the exact output if you try this?:
```
curl -v captain.captain.localhost
```

**Comment by juanferr** [comment]
thanks for following up, this is the reply. 

```
curl -v captain.captain.localhost
*   Trying 192.X.X.X:80...
* Connected to captain.captain.localhost (192.X.X.X) port 80 (#0)
> GET / HTTP/1.1
> Host: captain.captain.localhost
> User-Agent: curl/7.74.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 502 Bad Gateway
< Server: nginx
< Date: Fri, 30 Sep 2022 00:40:55 GMT
< Content-Type: text/html
< Content-Length: 150
< Connection: keep-alive
< 
<html>
<head><title>502 Bad Gateway</title></head>
<body>
<center><h1>502 Bad Gateway</h1></center>
<hr><center>nginx</center>
</body>
</html>
* Connection #0 to host captain.captain.localhost left intact

````

**Comment by githubsaturn** [comment]
Right after running curl run 
docker service logs captain-captain --since 5min

And

docker service logs captain-nginx --since 15m


The logs should tell whether the request made it to caprover container or not.

**Comment by juanferr** [comment]
```
aptain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:24.118 am    Waiting for 5 seconds for nginx reload to take into effect
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:29.121 am    Pruning containers...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:29.122 am    NGINX is fully set up and working...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:29.124 am    Captain Certbot is already running.. 
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:29.125 am    Updating Certbot service...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:29.133 am    executeCommand Container: captain-certbot certbot certificates --non-interactive 
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:29.714 am    **** Captain is initialized and ready to serve you! ****
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:34.132 am    Pruning containers...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:42.724 am    Retrying to get containerId for captain-certbot retry count:0
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:45.727 am    Retrying to get containerId for captain-certbot retry count:1
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:48.728 am    Retrying to get containerId for captain-certbot retry count:2
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:48.730 am    executeCommand Container: captain-certbot certbot certificates --non-interactive 
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:49.327 am    executeCommand Container: captain-certbot certbot renew --non-interactive 
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:49.800 am    Updating Load Balancer - renewAllCerts
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:49.800 am    Locking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:49.805 am    SUCCESS: UNLocking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:49.805 am    sendReloadSignal...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:28:49.807 am    Kill HUP Container: 6a22cc06e4f670a375fd96f73bee722b1b52f034c1cc37f13acc580748c5e308
captain-captain.1.ywyk5miybtzg@docker    | GET / 304 2.929 ms - -
captain-captain.1.ywyk5miybtzg@docker    | GET /static/js/2.a8870b67.chunk.js 304 0.705 ms - -
captain-captain.1.ywyk5miybtzg@docker    | GET /static/js/main.ee60e3c4.chunk.js 304 0.325 ms - -
captain-captain.1.ywyk5miybtzg@docker    | GET /icon-512x512.png 304 0.360 ms - -
captain-captain.1.ywyk5miybtzg@docker    | GET /light-theme.css 304 0.308 ms - -
captain-captain.1.ywyk5miybtzg@docker    | GET /dark-theme.css 304 0.298 ms - -
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:29:04.993 am    JsonWebTokenError: jwt must be provided
captain-captain.1.ywyk5miybtzg@docker    | JsonWebTokenError: jwt must be provided
captain-captain.1.ywyk5miybtzg@docker    |     at Object.module.exports [as verify] (/usr/src/app/node_modules/jsonwebtoken/verify.js:53:17)
captain-captain.1.ywyk5miybtzg@docker    |     at /usr/src/app/built/user/Authenticator.js:78:17
captain-captain.1.ywyk5miybtzg@docker    |     at new Promise (<anonymous>)
captain-captain.1.ywyk5miybtzg@docker    |     at Authenticator.decodeAuthToken (/usr/src/app/built/user/Authenticator.js:77:16)
captain-captain.1.ywyk5miybtzg@docker    |     at /usr/src/app/built/injection/Injector.js:43:14
captain-captain.1.ywyk5miybtzg@docker    |     at Layer.handle [as handle_request] (/usr/src/app/node_modules/express/lib/router/layer.js:95:5)
captain-captain.1.ywyk5miybtzg@docker    |     at trim_prefix (/usr/src/app/node_modules/express/lib/router/index.js:317:13)
captain-captain.1.ywyk5miybtzg@docker    |     at /usr/src/app/node_modules/express/lib/router/index.js:284:7
captain-captain.1.ywyk5miybtzg@docker    |     at Function.process_params (/usr/src/app/node_modules/express/lib/router/index.js:335:12)
captain-captain.1.ywyk5miybtzg@docker    |     at next (/usr/src/app/node_modules/express/lib/router/index.js:275:10)
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/system/versioninfo 200 3.483 ms - 62
captain-captain.1.ywyk5miybtzg@docker    | GET /light-theme.css 304 0.351 ms - -
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:29:04.997 am    JsonWebTokenError: jwt must be provided
captain-captain.1.ywyk5miybtzg@docker    | JsonWebTokenError: jwt must be provided
captain-captain.1.ywyk5miybtzg@docker    |     at Object.module.exports [as verify] (/usr/src/app/node_modules/jsonwebtoken/verify.js:53:17)
captain-captain.1.ywyk5miybtzg@docker    |     at /usr/src/app/built/user/Authenticator.js:78:17
captain-captain.1.ywyk5miybtzg@docker    |     at new Promise (<anonymous>)
captain-captain.1.ywyk5miybtzg@docker    |     at Authenticator.decodeAuthToken (/usr/src/app/built/user/Authenticator.js:77:16)
captain-captain.1.ywyk5miybtzg@docker    |     at /usr/src/app/built/injection/Injector.js:43:14
captain-captain.1.ywyk5miybtzg@docker    |     at Layer.handle [as handle_request] (/usr/src/app/node_modules/express/lib/router/layer.js:95:5)
captain-captain.1.ywyk5miybtzg@docker    |     at trim_prefix (/usr/src/app/node_modules/express/lib/router/index.js:317:13)
captain-captain.1.ywyk5miybtzg@docker    |     at /usr/src/app/node_modules/express/lib/router/index.js:284:7
captain-captain.1.ywyk5miybtzg@docker    |     at Function.process_params (/usr/src/app/node_modules/express/lib/router/index.js:335:12)
captain-captain.1.ywyk5miybtzg@docker    |     at next (/usr/src/app/node_modules/express/lib/router/index.js:275:10)
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/system/nginxconfig 200 1.646 ms - 62
captain-captain.1.ywyk5miybtzg@docker    | POST /api/v2/login 200 9.371 ms - 307
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/system/info 304 1.623 ms - -
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/system/versioninfo 200 315.347 ms - 153
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/apps/appDefinitions 304 1.545 ms - -
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/apps/appDefinitions 304 1.311 ms - -
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/apps/appDefinitions 304 1.180 ms - -
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/oneclick/repositories 304 1.701 ms - -
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/oneclick/template/list 304 289.084 ms - -
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:37:40.831 am    retrieving app at: https://oneclickapps.caprover.com/v4/apps/commento
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/oneclick/template/app?appName=commento&baseDomain=https:%2F%2Foneclickapps.caprover.com 200 108.421 ms - 1714
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/system/info 304 1.091 ms - -
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:02.306 am    Registering app started: commento
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:02.336 am    Building docker image. This might take a few minutes...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:02.388 am    Ensure service inited and Updated for: commento
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:02.388 am    Check if service is running: srv-captain--commento
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:02.390 am    Creating service srv-captain--commento with default image, we will update image later
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:02.394 am    Updating service srv-captain--commento with image img-captain-commento:0
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:04.402 am    Updating Load Balancer - ServiceManager
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:04.403 am    Locking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:04.407 am    SUCCESS: UNLocking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:04.407 am    sendReloadSignal...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:04.409 am    Kill HUP Container: 6a22cc06e4f670a375fd96f73bee722b1b52f034c1cc37f13acc580748c5e308
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:04.429 am    AppName is saved: commento
captain-captain.1.ywyk5miybtzg@docker    | POST /api/v2/user/apps/appDefinitions/register 200 2124.690 ms - 61
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/apps/appDefinitions 200 1.174 ms - 4296
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:04.510 am    Updating app started: commento
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:04.515 am    Ensure service inited and Updated for: commento
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:04.515 am    Check if service is running: srv-captain--commento
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:04.517 am    Service is already running: srv-captain--commento
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:04.517 am    Updating service srv-captain--commento with image img-captain-commento:0
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:06.529 am    Updating Load Balancer - ServiceManager
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:06.529 am    Locking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:06.533 am    SUCCESS: UNLocking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:06.533 am    sendReloadSignal...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:06.535 am    Kill HUP Container: 6a22cc06e4f670a375fd96f73bee722b1b52f034c1cc37f13acc580748c5e308
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:06.549 am    Updating Load Balancer - ServiceManager
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:06.550 am    Locking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:06.553 am    SUCCESS: UNLocking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:06.553 am    sendReloadSignal...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:06.555 am    Kill HUP Container: 6a22cc06e4f670a375fd96f73bee722b1b52f034c1cc37f13acc580748c5e308
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:06.569 am    AppName is updated: commento
captain-captain.1.ywyk5miybtzg@docker    | POST /api/v2/user/apps/appDefinitions/update 200 2059.925 ms - 69
captain-captain.1.ywyk5miybtzg@docker    | POST /api/v2/user/apps/appData/commento 500 556.275 ms - 21
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:07.339 am    Error: (HTTP code 404) unexpected - manifest for registry.gitlab.com/commento/commento@sha256:e792ac5c384ca7d45bfc228ef6ec3084a54af98847ee75a8aac7aa720f4f87a2 not found: manifest unknown: manifest unknown 
captain-captain.1.ywyk5miybtzg@docker    | Error: (HTTP code 404) unexpected - manifest for registry.gitlab.com/commento/commento@sha256:e792ac5c384ca7d45bfc228ef6ec3084a54af98847ee75a8aac7aa720f4f87a2 not found: manifest unknown: manifest unknown 
captain-captain.1.ywyk5miybtzg@docker    |     at /usr/src/app/node_modules/docker-modem/lib/modem.js:315:17
captain-captain.1.ywyk5miybtzg@docker    |     at IncomingMessage.<anonymous> (/usr/src/app/node_modules/docker-modem/lib/modem.js:342:9)
captain-captain.1.ywyk5miybtzg@docker    |     at IncomingMessage.emit (events.js:412:35)
captain-captain.1.ywyk5miybtzg@docker    |     at endReadableNT (internal/streams/readable.js:1334:12)
captain-captain.1.ywyk5miybtzg@docker    |     at processTicksAndRejections (internal/process/task_queues.js:82:21)
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:07.402 am    Pruning containers...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:09.529 am    Pruning containers...
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/apps/appDefinitions 200 1.169 ms - 4673
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/apps/appDefinitions 304 1.193 ms - -
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:28.387 am    Deleting app started: commento
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:28.387 am    Removing service for: commento
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:28.388 am    Check if service is running: srv-captain--commento
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:30.405 am    Updating Load Balancer - ServiceManager
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:30.406 am    Locking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:30.410 am    SUCCESS: UNLocking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:30.410 am    sendReloadSignal...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:30.411 am    Kill HUP Container: 6a22cc06e4f670a375fd96f73bee722b1b52f034c1cc37f13acc580748c5e308
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:30.466 am    AppName is deleted: commento
captain-captain.1.ywyk5miybtzg@docker    | POST /api/v2/user/apps/appDefinitions/delete 200 2079.958 ms - 55
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/apps/appDefinitions 200 1.073 ms - 3794
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/oneclick/repositories 304 0.948 ms - -
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/oneclick/template/list 304 78.006 ms - -
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:38:49.918 am    retrieving app at: https://oneclickapps.caprover.com/v4/apps/dokuwiki
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/oneclick/template/app?appName=dokuwiki&baseDomain=https:%2F%2Foneclickapps.caprover.com 200 133.924 ms - 2367
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/system/info 304 1.655 ms - -
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:16.057 am    Registering app started: doku
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:16.073 am    Building docker image. This might take a few minutes...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:16.124 am    Ensure service inited and Updated for: doku
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:16.124 am    Check if service is running: srv-captain--doku
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:16.126 am    Creating service srv-captain--doku with default image, we will update image later
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:16.129 am    Updating service srv-captain--doku with image img-captain-doku:0
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:18.134 am    Updating Load Balancer - ServiceManager
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:18.134 am    Locking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:18.139 am    SUCCESS: UNLocking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:18.139 am    sendReloadSignal...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:18.143 am    Kill HUP Container: 6a22cc06e4f670a375fd96f73bee722b1b52f034c1cc37f13acc580748c5e308
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:18.186 am    AppName is saved: doku
captain-captain.1.ywyk5miybtzg@docker    | POST /api/v2/user/apps/appDefinitions/register 200 2129.743 ms - 61
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/apps/appDefinitions 200 1.079 ms - 4287
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:18.211 am    Updating app started: doku
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:18.221 am    Ensure service inited and Updated for: doku
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:18.222 am    Check if service is running: srv-captain--doku
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:18.223 am    Service is already running: srv-captain--doku
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:18.223 am    Updating service srv-captain--doku with image img-captain-doku:0
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:20.232 am    Updating Load Balancer - ServiceManager
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:20.232 am    Locking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:20.236 am    SUCCESS: UNLocking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:20.236 am    sendReloadSignal...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:20.238 am    Kill HUP Container: 6a22cc06e4f670a375fd96f73bee722b1b52f034c1cc37f13acc580748c5e308
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:20.254 am    Updating Load Balancer - ServiceManager
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:20.254 am    Locking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:20.258 am    SUCCESS: UNLocking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:20.258 am    sendReloadSignal...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:20.260 am    Kill HUP Container: 6a22cc06e4f670a375fd96f73bee722b1b52f034c1cc37f13acc580748c5e308
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:20.281 am    AppName is updated: doku
captain-captain.1.ywyk5miybtzg@docker    | POST /api/v2/user/apps/appDefinitions/update 200 2070.710 ms - 69
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:21.133 am    Pruning containers...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:23.231 am    Pruning containers...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:29.947 am    Ensure service inited and Updated for: doku
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:29.947 am    Check if service is running: srv-captain--doku
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:29.948 am    Service is already running: srv-captain--doku
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:29.949 am    Updating service srv-captain--doku with image bitnami/dokuwiki:20200729-debian-10
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:31.955 am    Updating Load Balancer - ServiceManager
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:31.955 am    Locking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:31.960 am    SUCCESS: UNLocking NGINX configuration reloading...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:31.960 am    sendReloadSignal...
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:31.961 am    Kill HUP Container: 6a22cc06e4f670a375fd96f73bee722b1b52f034c1cc37f13acc580748c5e308
captain-captain.1.ywyk5miybtzg@docker    | POST /api/v2/user/apps/appData/doku 200 11714.834 ms - 55
captain-captain.1.ywyk5miybtzg@docker    | GET /api/v2/user/apps/appDefinitions 200 1.051 ms - 4871
captain-captain.1.ywyk5miybtzg@docker    | September 29th 2022, 5:39:34.954 am    Pruning containers...
root@docker:/etc# docker service logs captain-nginx --since 15m
captain-nginx.1.slubdkpagmej@docker    | 2022/09/30 01:12:04 [error] 31#31: *10 connect() failed (113: No route to host) while connecting to upstream, client: 192.X.X.X, server: captain.captain.localhost, request: "GET / HTTP/1.1", upstream: "http://10.0.1.2:3000/", host: "captain.captain.localhost"
```

**Comment by juanferr** [comment]
tf it helps this is the bitwarden log from caprover says it launched `Rocket has launched from http://0.0.0.0:80` is that correct?
```

2022-09-29T05:14:27.841596641Z | Starting Vaultwarden |
2022-09-29T05:14:27.841600099Z | Version 1.21.0 |
2022-09-29T05:14:27.841602427Z |--------------------------------------------------------------------|
2022-09-29T05:14:27.841604574Z | This is an *unofficial* Bitwarden implementation, DO NOT use the |
2022-09-29T05:14:27.841606695Z | official channels to report bugs/features, regardless of client. |
2022-09-29T05:14:27.841608793Z | Send usage/configuration questions or feature requests to: |
2022-09-29T05:14:27.841610895Z | https://bitwardenrs.discourse.group/ |
2022-09-29T05:14:27.841612965Z | Report suspected bugs/issues in the software itself at: |
2022-09-29T05:14:27.841615040Z | https://github.com/dani-garcia/vaultwarden/issues/new |
2022-09-29T05:14:27.841617171Z \--------------------------------------------------------------------/
2022-09-29T05:14:27.841619345Z
2022-09-29T05:14:27.841789369Z [INFO] No .env file found.
2022-09-29T05:14:27.841795134Z
2022-09-29T05:14:27.857210073Z [2022-09-29 05:14:27.856][parity_ws][INFO] Listening for new connections on 0.0.0.0:3012.
2022-09-29T05:14:27.857222578Z [2022-09-29 05:14:27.857][start][INFO] Rocket has launched from http://0.0.0.0:80
```

**Comment by githubsaturn** [comment]
Okay, here is the problem:
```
root@docker:/etc# docker service logs captain-nginx --since 15m
captain-nginx.1.slubdkpagmej@docker    | 2022/09/30 01:12:04 [error] 31#31: *10 connect() failed (113: No route to host) while connecting to upstream, client: 192.X.X.X, server: captain.captain.localhost, request: "GET / HTTP/1.1", upstream: "http://10.0.1.2:3000/", host: "captain.captain.localhost"
```

Your nginx container cannot connect to your CapRover container. Most likely you're missing some kernel modules that Docker uses for network overlay

**Comment by juanferr** [comment]
interesting, im using the latest docker engine

```
Client: Docker Engine - Community
 Version:           20.10.18
 API version:       1.41
 Go version:        go1.18.6
 Git commit:        b40c2f6
 Built:             Thu Sep  8 23:12:08 2022
 OS/Arch:           linux/amd64
 Context:           default
```

**Comment by githubsaturn** [comment]
It's not the docker. It's your OS. 

You can simply shell into your nginx container and try to see if you can reach out to captain- captain container. You'll see that it'll fail to resolve the DNS

**Comment by githubsaturn** [comment]
You can read more details about this issue here:
https://github.com/caprover/caprover/issues/1252#issuecomment-964248406

TLDR: it's an issue with Docker, not CapRover, you can reproduce it without CapRover. Feel free to file an issue on Docker's github page and link this issue.
